src/main/cpp/src/json_parser.cuh

```diff
@@ -998,7 +985,8 @@ class json_parser {
   __device__ bool try_skip_unicode(char const*& str_pos,
                                    char const*& to_match_str_pos,
                                    char const* const to_match_str_end,
-                                   char*& copy_dest)
+                                   char*& copy_dest,
+                                   int& unescped_string_utf8_bytes)
```

Comment:
Please pass in `escped_string_utf8_bytes` and update.
I found an existing bug which is not introduced in this PR.
Spark behaviors:
```
    // use unescaping logic
    get_json_object("['\\u4e2d\\u56FD']", "$")    : ["中国"]

    // use escaping logic
    get_json_object("'\\u4e2d\\u56FD'", "$")      : 中国
```
For the above 2 cases, both need to update the utf8_bytes for both `ESCAPE`  and `UNESCAPE` mode.
Please update test case `getJsonObjectTest_Escape`
```
    String JSON6 = "['\\u4e2d\\u56FD\\\"\\'\\\\\\/\\b\\f\\n\\r\\t\\b']";
    String expectedStr6 = "中国\\\"'\\\\/\\b\\f\\n\\r\\t\\b";
```

And please update the PR description to describe this bug.

Reply:
Nice catch! done.